Tests/UnitTests/Core/Geometry/BlueprintApiTests.cpp

```diff
@@ -276,89 +272,100 @@ BOOST_AUTO_TEST_CASE(NodeApiTestContainers) {
     mat.configureFace(NegativeDisc, {AxisR, Bound, 15}, {AxisPhi, Bound, 25});
     mat.configureFace(PositiveDisc, {AxisR, Bound, 15}, {AxisPhi, Bound, 25});
 
-    mat.addCylinderContainer("Detector", AxisDirection::AxisR, [&](auto& det) {
-      det.addCylinderContainer("Pixel", AxisDirection::AxisZ, [&](auto& cyl) {
-        cyl.setAttachmentStrategy(VolumeAttachmentStrategy::Gap)
-            .setResizeStrategy(VolumeResizeStrategy::Gap);
-
-        cyl.addCylinderContainer(
-            "PixelNegativeEndcap", AxisDirection::AxisZ, [&](auto& ec) {
-              ec.setAttachmentStrategy(VolumeAttachmentStrategy::Gap);
-
-              auto makeLayer = [&](const Transform3& trf, auto& layer) {
-                std::vector<std::shared_ptr<Surface>> surfaces;
-                auto layerSurfaces = makeFan(trf, 300_mm, 10, 2_mm);
-                std::copy(layerSurfaces.begin(), layerSurfaces.end(),
-                          std::back_inserter(surfaces));
-                layerSurfaces = makeFan(trf, 500_mm, 16, 2_mm);
-                std::copy(layerSurfaces.begin(), layerSurfaces.end(),
-                          std::back_inserter(surfaces));
-
-                layer.setSurfaces(surfaces)
-                    .setLayerType(LayerBlueprintNode::LayerType::Disc)
-                    .setEnvelope(ExtentEnvelope{{
-                        .z = {5_mm, 5_mm},
-                        .r = {10_mm, 20_mm},
-                    }})
-                    .setTransform(base);
-              };
-
-              ec.addLayer("PixelNeg1", [&](auto& layer) {
-                makeLayer(base * Translation3{Vector3{0, 0, -700_mm}}, layer);
+    mat.addCylinderContainer(
+        "Detector", AxisDirection::AxisR,
+        [&](Experimental::CylinderContainerBlueprintNode& det) {
```

Comment:
Debatable if the type should be written out here, I think.

Reply:
uff the diff is huge - let me revert that. I did this mainly because I could not understand the parameter type